extensions/functions_string.yaml

```diff
@@ -163,3 +163,355 @@ scalar_functions:
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: The input string.
+          - value: "string"
+            name: "substring"
+            description: The substring to count.
+        return: i8
+      - args:
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: The input string.
+          - value: "varchar<L1>"
+            name: "substring"
+            description: The substring to count.
+        return: i8
+  -
+    name: regexp_count_substring
+    description: Returns the number of times a regular expression pattern occurs in an input string.
+    impls:
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "string"
+            name: "input"
+            description: Input string.
+          - value: "string"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: i8
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "varchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "varchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: i8
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "fixedchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: i8
+  -
+    name: regexp_matches
+    description: >-
+      Returns true if part of the input string matches the regular expression pattern.
+    impls:
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "string"
+            name: "input"
+            description: Input string.
+          - value: "string"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: "BOOLEAN"
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "varchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "varchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: "BOOLEAN"
+      - args:
+          - options: [ CASE_SENSITIVE, CASE_INSENSITIVE ]
+            required: false
+          - value: "fixedchar<L1>"
+            name: "input"
+            description: Input string.
+          - value: "fixedchar<L1>"
+            name: "pattern"
+            description: The regular expression pattern.
+        return: "BOOLEAN"
+  - name: replace
+    description: >-
+      Replace any occurrence of the substring with the replacement string
```

Comment:
"any" as in one of them or as in all of them? If the former, which one?

I think if you're being properly pedantic about what this says it unambiguously means all occurrences, but I'd rather see it written as "replaces all occurrences" to avoid confusion.

Also, nit: missing `.` at the end of the sentence.